src/parser/parser_driver.cpp

```diff
@@ -433,51 +435,96 @@ void ParserDriver::defineGrammar()
-				_lastRuleLocation = currentFileContext()->getLocation();
-				_lastRuleTokenStream = currentFileContext()->getTokenStream();
-				return {};
-			}, "ID", [&](auto&& args) -> Value {
-				const std::string& name_text = args[3].getTokenIt()->getString();
-				if (ruleExists(name_text))
-					error_handle(args[3].getTokenIt()->getLocation(), "Redefinition of rule '" + args[3].getTokenIt()->getString() + "'");
-				args[3].getTokenIt()->setType(TokenType::RULE_NAME);
-				args[3].getTokenIt()->setValue(std::make_shared<ValueSymbol>(name_text, Expression::Type::Bool));
-				return {};
-			},
-			"tags", "LCB", "metas", "strings", "condition", "RCB", [&](auto&& args) -> Value {
-				std::optional<TokenIt> mod_private = {};
-				std::optional<TokenIt> mod_global = {};
-				const std::vector<TokenIt> mods = std::move(args[0].getMultipleTokenIt());
-				for (const auto &token: mods)
-				{
-					if (token->getType() == TokenType::GLOBAL)
-					{
-						if (mod_global.has_value())
-							error_handle(token->getLocation(), "Duplicated global rule modifier");
-						mod_global = token;
-					}
-					else if (token->getType() == TokenType::PRIVATE)
-					{
-						if (mod_private.has_value())
-							error_handle(token->getLocation(), "Duplicated private rule modifier");
-						mod_private = token;
-					}
-				}
-				TokenIt name = args[3].getTokenIt();
-				const std::vector<TokenIt> tags = std::move(args[5].getMultipleTokenIt());
-				args[6].getTokenIt()->setType(TokenType::RULE_BEGIN);
-				std::vector<Meta> metas = std::move(args[7].getMetas());
-				std::shared_ptr<Rule::StringsTrie> strings = std::move(args[8].getStringsTrie());
-				Expression::Ptr condition = std::move(args[9].getExpression());
-				args[10].getTokenIt()->setType(TokenType::RULE_END);
-
-				addRule(Rule(_lastRuleTokenStream, name, std::move(mod_private), std::move(mod_global),
-					std::move(metas), std::move(strings), std::move(condition), std::move(tags)));
-				return {};
-			});
+  auto const common_last_rule = [&](auto&&) -> Value {
+  	_lastRuleLocation = currentFileContext()->getLocation();
+	  _lastRuleTokenStream = currentFileContext()->getTokenStream();
+		return {};
+	};
+
+  auto const common_rule_init = [&](auto&& args) -> Value {
+		const std::string& name_text = args[3].getTokenIt()->getString();
+		if (ruleExists(name_text))
+		  error_handle(args[3].getTokenIt()->getLocation(), "Redefinition of rule '" + args[3].getTokenIt()->getString() + "'");
+		args[3].getTokenIt()->setType(TokenType::RULE_NAME);
+		args[3].getTokenIt()->setValue(std::make_shared<ValueSymbol>(name_text, Expression::Type::Bool));
+		return {};
+	};
+
+  if (_features & Features::AvastOnly)
+    _parser.rule("rule") // {}
+		  .production(
+			  "rule_mods", "RULE", common_last_rule, "ID", common_rule_init, "tags", "LCB", "metas", "strings", "variables", "condition" , "RCB", [&](auto&& args) -> Value {
+	        std::optional<TokenIt> mod_private = {};
```

Comment:
Check if you are using tabs/spaces consistently. Some lines are aligned differently than others.